### DISCUSSION TITLE
Secure it with an account

### DISCUSSION THREAD

**Discussion by steemsjo** [blakeblackshear/frigate#3508] [Ideas]
I was in the process of exposing Frigate outside of my LAN but want to have it shielded with a MFA account. 

Is there any ideas or plans about securing it with a ACL? Would be nice to be able to expose only certain cameras to certain users etc. …

**Comment by NickM-27** [comment]
There's been similar questions in the past and the answer has been probably not. With frigate as a docker container outside services / proxies can easily manage this.

For example I have NGINX -> oauthproxy -> frigate so to access frigate from WAN it requires signing in with Google. For frigate to re-implement this itself would be a lot of duplication. 

The exposing of certain cameras I think would be more involved but not impossible, and perhaps some analogous features outside of authentication could be added in the future to make that easier.

**Reply by steemsjo** [reply]
Didn’t knew similar things were asked before. 
But your explanation makes sense! 

Using Cloudflare on Home Assistant as a add-on to expose Frigate (docker container on another server) to WAN. 
But it’s open now for everybody. Was looking into securing it. All the other services that I expose have their own auth build in. That’s the reason I was aiming for the same on Frigate.

**Comment by aisbergde** [comment]
similar #1580 - is there an easy way to get some basic authentication?

I installed frigate in docker on a Synology NAS. And I want to grant external access. I can easily grant external access using the Synology build in reverse proxy. it works fine with other apps, which have their own user management. And I can't find, how to connect the Synology reverse proxy with a user authentication. And there are other statements: "the docker container should care about authentication, the reverse proxy is just a reverse proxy".

I am not a network nerd, I was able to set up simple docker containers. I also use home assistant for many years. But now I moved it from a Raspberry Pi into the docker on Synology NAS and I even don't know, how to install custom components or HACS. But it looks like, frigate could be used also without HA very well.

**Comment by aisbergde** [comment]
I have defined WireGuard VPN tunnels. This works directly in my router (FritzBox). So I have found a good solution.
